modules/codegen/src/main/kotlin/jp/co/soramitsu/iroha2/codegen/generator/scale.kt

```diff
@@ -83,7 +82,10 @@ fun resolveScaleReadImpl(type: Type): CodeBlock {
             }
         }
         is OptionType -> {
-            CodeBlock.of("reader.readNullable(%T)", withoutGenerics(resolveKotlinType(type)))
+            when (type.name) {
+                "Option<u32>", "Option<u16>" -> CodeBlock.of("reader.readNullable()")
```

Comment:
It is no sae to rely on name in the schema, due it can change, better to rely on resolved type
```suggestion
            when(type.innerType.requireValue()) {
                is U32Type, U16Type -> CodeBlock.of("reader.readNullable()")
```
Please update here and below